### PR TITLE
fix: never leave a page blank when there is no usable GPU

### PR DESCRIPTION
**Pull request by pottekkat**
#### Changes

On a machine without a usable GPU, the whole page could stay blank. The offscreen document brings TensorFlow.js up on WebGL, and when the model warm-up timed out there, the old code called `setBackend('wasm')` on the live engine. `withTimeout` doesn't cancel the GPU work it gave up on, so the engine waited on it forever, and because backend bring-up runs on the serialised op chain every queued image was stuck behind it. Images are hidden from the moment they're queued and revealed only when a verdict comes back, so they were hidden for the life of the page. Reproduced locally with a 330s run where every image was still `data-nsfw-filter-status="processing"` and `visibility: hidden`.

Three changes:

- The content script now owns a 60s deadline per URL. If no verdict arrives, the image is revealed instead of staying hidden. Showing an unclassified image is a bad outcome; a permanently blank page is a worse one, and it's how the filter already degrades when the background worker is down. A reply that arrives after we gave up is dropped rather than resolving twice, matched by request rather than by URL so it can't settle a later request for the same image, and giving up also stops the reconnect timer. The existing "background worker is down" path had the same shape of bug: it only resolved the first waiter, so duplicate `` elements sharing a src stayed hidden.
- The offscreen document no longer switches backends in place, on any WebGL failure. It writes what it needs to `sessionStorage` and reloads itself, so WASM comes up in a clean realm with no orphaned GPU work, and the recorded restart stops a bad GPU turning into a reload loop. The service worker doesn't re-push settings to a document it didn't start, so the strictness, model, and logging state ride along in that same record. Alongside that, `setBackend()` returns a boolean that was being ignored, the `tf.ready()` after it was redundant, and `currentBackend` was set before the backend was known to be live.
- `PredictionQueue.onFailure` no longer caches the URL as safe. A failure means no verdict came back, usually the model being unavailable, so caching `false` kept serving "safe" for that URL for the rest of the session.

#### Testing

- Unit tests for the deadline, the reconnect give-up, and stale replies (`test/unit/Filter.test.ts`), and for the failure cache (`test/unit/PredictionQueue.test.ts`). All six fail against `master` and pass here.
- A new e2e suite (`test/e2e/noGpuFallback.test.js`) runs the extension in a second browser launched with GL off, which is the real no-GPU path. It asserts the page settles inside 40s, below the content script's deadline, so a page that only recovered by failing open still fails the test, and it reads the offscreen document's restart record to prove it got to WASM by restarting once rather than by switching in place. The harness launches that browser in `globalSetup` because test files can't launch Puppeteer themselves.
- Also verified in a real browser with a forced WebGL warm-up failure, the case the flags can't produce: the offscreen document reloaded, came back up on WASM with the settings intact, and classified normally.

#### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

## Summary by CodeRabbit

- **Bug Fixes**
  - Failed image analyses no longer produce cached “safe” results.
  - Images are revealed after a timeout when analysis cannot complete.
  - Late or stale analysis responses no longer affect newer requests.
  - Recovery is improved when background processing or reconnection attempts fail.

- **Reliability**
  - Improved WebGL initialization and automatic fallback to WASM when GPU support is unavailable.
  - Processing can restart more reliably while preserving settings.

- **Tests**
  - Added coverage for timeout handling, recovery, stale responses, restart persistence, and no-GPU operation.

### CODE REVIEW

File: src/content/Filter/Filter.ts [hunk 1]
```diff
@@ -27,52 +42,89 @@ export class Filter implements IFilter {
...
+      if (pending === undefined) return
+
       console.warn(`[NSFW-Filter] Background worker is down, marked as visible ${request.url}`)
-      this.requestQueue.delete(request.url)
+      for (const { resolve } of pending.waiters) {
+        resolve(new PredictionResponse(false, request.url, 'Background worker doesn\'t working'))
+      }
     } else {
-      request.reconectTimer = window.setTimeout(() => this._requestToAnalyzeImage(request, resolve), 500)
+      request.reconectTimer = window.setTimeout(() => this._requestToAnalyzeImage(request), 500)
```
[the remark this answers is not in the document]
Comment (pottekkat) [reply src/content/Filter/Filter.ts:131]: Done in 3935be9. `_handleBackgroundErrors` now returns early unless the request is still the one on the queue, and there is a unit test for the requeue case (`drops a late runtime error belonging to an abandoned request`).

File: test/e2e/noGpuFallback.test.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      [...document.images].every(image => {
+        const status = image.getAttribute('data-nsfw-filter-status')
+        return status !== null && status !== 'processing'
+      }),
+    { timeout: SETTLE_TIMEOUT, polling: 500 })
+
+    const hidden = await page.evaluate(() =>
+      [...document.images].filter(image => getComputedStyle(image).visibility === 'hidden').length
+    )
+    expect(hidden).toBe(0)
```
[the remark this answers is not in the document]
Comment (pottekkat) [reply test/e2e/noGpuFallback.test.js:67]: Done in 3935be9.

File: test/e2e/noGpuFallback.test.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    expect(offscreen).toBeDefined()
+
+    const session = await offscreen.createCDPSession()
+    const { result } = await session.send('Runtime.evaluate', {
+      expression: `sessionStorage.getItem('${RESTART_KEY}')`,
+      returnByValue: true
+    })
+    await session.detach()
+
+    expect(result.value).not.toBeNull()
```
[the remark this answers is not in the document]
Comment (pottekkat) [reply test/e2e/noGpuFallback.test.js]: Addressed in 3935be9, differently. The record is asserted for shape and the offscreen document now has to answer a classification without an error, which is what proves WASM actually came up: the background reports a failed prediction as `false`, so a settled page cannot tell inference from failing open. The settings the record carries are covered by `test/unit/restartState.test.ts` instead. Seeding them in e2e is not possible, the document restarts before any test code can run. Reload count is `expect(navigations).toEqual(['reload'])`.
